### ISSUE TITLE
UI improvement (epic): Refactor tables to grids across areas for more responsive and flexible design 

### ISSUE THREAD

**Issue by cdrage** [containers/podman-desktop#885]
### Epic domain

Unfortunately we're starting to run into limitations of the tailwindcss table feature:
* Unable to adjust width of columns (see: https://github.com/containers/podman-desktop/issues/506 issue)
* Only two settings for table (auto and manual)
*. Not responsive design (information gets cut off) when viewing as a smaller screen

As an epic, we should refactor using tables for grids instead which will enable us to:
* Add sorting on each column
* Be able to adjust the width of each column
* More responsive design when doing smaller / larger screens

More research will need to be done if this can be accomplished by either using a grid layout or some form of plugin / added dependancy to fill the limitations of using tailwindcss tables.

Issue breakdown:
- [x] #4545 (adds ability to sort and proportional columns)
- [x] #4839 Apply component to Volumes page (including column width, initial sorting)
- [x] #4840 Apply component to Images page (including column width, initial sorting)
- [x] #4842 Apply component to Pods page (including column width, initial sorting)
- [x] #4843 Apply component to Containers page (including column width, initial sorting)
- [ ] Ability to adjust column width

**Comment by nichjones1** [comment]
@deboer-tim are the items in this epic part of the table work that you are working on?

**Comment by deboer-tim** [comment]
Yes. The table component issue #4545 will add ability to sort tables (the first issue above), but we still need other child issues of this epic to use the component for all pages and provide comparators to make each column sortable.

The table component will also provide more responsive design (e.g. columns can be fixed or proportional width), which also need to be applied to each screen).

So: one issue per page (Pods, Containers, etc) to apply the table component, column width, and sorting. I mostly have Volumes and Images done so those should be soon.

Being able to adjust column width is not covered yet, that will need to be a separate issue, but hopefully can be done once for all tables after the rest is done.

I added a section to the description with this breakdown.

**Comment by deboer-tim** [comment]
Working on final issue: containers page.

**Comment by deboer-tim** [comment]
The main work to use a common, more responsive table is complete, and we have sorting for most columns. @cdrage it seems like the main remaining issue here was the ability to resize columns but there's no actual issue opened. Do you want to open one? Not sure there is a benefit to keeping this open as an epic.

**Comment by cdrage** [comment]
> The main work to use a common, more responsive table is complete, and we have sorting for most columns. @cdrage it seems like the main remaining issue here was the ability to resize columns but there's no actual issue opened. Do you want to open one? Not sure there is a benefit to keeping this open as an epic.

IMHO the resizing hasn't been discussed and to be honest I am not sure if I want that feature anymore haha.

I consider this closed as all the improvements have been completed to the new table code!
